src/charm.py

```diff
@@ -688,6 +740,9 @@ def _get_configs_per_leader(self, relation: Relation) -> Dict[str, Any]:
             redirect_https=data.get("redirect-https", False),
             strip_prefix=data.get("strip-prefix", False),
             external_host=self.external_host,
+            enable_experimental_forward_auth=self._is_forward_auth_enabled,
```

Comment:
I'd make `experimental_forward_auth` an init argument to `self.traefik`, since we seem to be passing it in multiple times.

Reply:
(next to `tls_enabled: same pattern there)